### PR TITLE
[FEATURE BRANCH] Auth address tiers

### PR DESCRIPTION
**Pull request by Ation**
**Do not merge till feature ready**

### CODE REVIEW

File: BlocksettleNetworkingLib/AuthAddressManager.cpp [hunk 1]
```diff
@@ -91,8 +91,19 @@ bool AuthAddressManager::setup()
          return;
       }
       if (GetState(address) != state) {
-         logger_->info("Address verification {} for {}", to_string(state), address.display());
-         SetState(address, state);
+         logger_->info("Address verification on chain {} for {}", to_string(state), address.display());
+
+         if (state == AddressVerificationState::NotSubmitted) {
+            const auto &submittedAddresses = celerClient_->GetSubmittedAuthAddressSet();
```
Comment (goatpig) [inline BlocksettleNetworkingLib/AuthAddressManager.cpp:97]: what is celer's involvement with auth addresses?

File: BlocksettleNetworkingLib/AuthAddressManager.cpp [hunk 2]
```diff
@@ -91,8 +91,19 @@ bool AuthAddressManager::setup()
...
+            if (submittedAddresses.find(address.display()) != submittedAddresses.end()) {
+               SetState(address, AddressVerificationState::Submitted);
+            } else {
+               SetState(address, state);
```
Comment (goatpig) [inline BlocksettleNetworkingLib/AuthAddressManager.cpp:101]: Looking at SetState, this call will return with no effect. Why call the method at all in a dedicated else branch? This suggests auth addresses that are presented are being tracked even when they're invalid. It's a bit confusing.

File: BlocksettleNetworkingLib/AuthAddressManager.cpp (hunk 1 above)
Comment (Ation) [reply BlocksettleNetworkingLib/AuthAddressManager.cpp:97]: we have list of "submitted auth addresses" saved as user property on celer. Since there is no way for terminal to know if address was submitted or not if terminal is clean setup and wallet was imported.

File: BlocksettleNetworkingLib/AuthAddressManager.cpp (hunk 2 above)
Comment (Ation) [reply BlocksettleNetworkingLib/AuthAddressManager.cpp:101]: state in this method - on-chain validation result.
SetState sort of auth address cache.

File: BlocksettleNetworkingLib/AuthAddressManager.cpp (hunk 1 above)
Comment (goatpig) [reply BlocksettleNetworkingLib/AuthAddressManager.cpp:97]: why is this done by celer and not the PB storage DB or genoa is what im asking

**Comment by goatpig** [comment]
common submodule wasn't updated in PR to the PB repo btw.

File: BlocksettleNetworkingLib/AuthAddressManager.cpp (hunk 2 above)
Comment (goatpig) [reply BlocksettleNetworkingLib/AuthAddressManager.cpp:101]: Let me rephrase: the call to SetState at line 101 will have no effect. Why call it at all?

File: BlocksettleNetworkingLib/AuthAddressManager.cpp (hunk 1 above)
Comment (Ation) [reply BlocksettleNetworkingLib/AuthAddressManager.cpp:97]: because it was done 4 years ago

File: BlocksettleNetworkingLib/AuthAddressManager.cpp (hunk 2 above)
Comment (Ation) [reply BlocksettleNetworkingLib/AuthAddressManager.cpp:101]: It will change address state from InProgress to NotSubmitted
